src/app/SignalR/Services/ChangeStreamBackgroundService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Common.DAL.Documents.User;
+using Common.DAL.Interfaces;
+using Microsoft.Extensions.Hosting;
+using MongoDB.Driver;
+using SignalR.Hubs;
+
+namespace SignalR.Services
+{
+    public class ChangeStreamBackgroundService : BackgroundService
+    {
+        private readonly IMongoCollection<User> _users;
+        private readonly IUserHubContext _userHubContext;
+
+        public ChangeStreamBackgroundService(IDbContext dbContext, IUserHubContext userHubContext)
+        {
+            _users = dbContext.Users;
+            _userHubContext = userHubContext;
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            var options = new ChangeStreamOptions() { FullDocument = ChangeStreamFullDocumentOption.UpdateLookup };
+            var pipeline = new EmptyPipelineDefinition<ChangeStreamDocument<User>>()
+                .Match(x => x.OperationType == ChangeStreamOperationType.Replace || x.OperationType == ChangeStreamOperationType.Update);
+
+            using var cursor = _users.Watch(pipeline, options);
+            await cursor.ForEachAsync(async document =>
+            {
+                await _userHubContext.SendUpdateAsync(document.FullDocument);
+            });
```

Comment:
You can pass `stoppingToken` to `Watch` and `ForEachAsync` methods.

Reply:
@yarmolovichalex, Yes, but I tested this feature with CancellationTokenSource(1000).Token and it doesn't work...

Reply:
Hm, looks like I tried to pass only to Watch. Passing to Watch and to ForEachAsync works, thanks!